### ISSUE TITLE
Add retries on file sync failure

### ISSUE THREAD

**Issue by NanobotZ** [superzanti/ServerSync#290]
#### Serversync Version: 
4.2.0

#### Minecraft Version: 
doesn't matter

#### Issue: 
According to this line, it doesn't matter whether file updates are run successfully or not, the program doesn't inform the user that something went wrong while processing nor it doesn't try to retry the update action again later, as the boolean value returned by this function isn't handled anywhere.
https://github.com/superzanti/ServerSync/blob/c2d9371054bf23fffa6a30eb2f69225635546ae2/src/main/java/com/superzanti/serversync/client/Mode2Sync.java#L38

I don't think this is a good behavior, as client users with spotty internet connection (I know two that run into this problem) may not even notice that something went wrong while syncing with the server, especially if running in no-GUI mode.

**Comment by rheimus** [comment]
Yep, retries ended up being removed at some point. Makes sense to add them back in with some sort of exit handling/ui for it as well.
